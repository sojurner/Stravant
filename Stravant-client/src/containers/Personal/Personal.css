.personal-stats {
  display: flex;
  justify-content: center; }

path {
  cursor: pointer; }

.personal-stats {
  position: absolute; }

.legend {
  background-color: #f9f9f9;
  border: 1px solid #e5e5e5;
  border-radius: 12px;
  font-size: 0.8em;
  padding: 2rem; }

.pie-chart-wrap {
  display: flex;
  justify-content: center; }

li {
  display: flex;
  left: 0.6rem; }

.data-details {
  position: absolute;
  top: 12rem;
  left: 23rem;
  font-family: daybold;
  font-size: 1.5rem;
  color: #ffc1b8;
  letter-spacing: 2px;
  font-size: 1.5rem;
  background: #c5696a;
  border-radius: 0.4em;
  padding: 1rem;
  opacity: 0.9; }

.data-details:after {
  content: '';
  position: absolute;
  left: 0;
  top: 50%;
  width: 0;
  height: 0;
  border: 21px solid transparent;
  border-right-color: #c5696a;
  border-left: 0;
  border-bottom: 0;
  margin-top: -10.5px;
  margin-left: -21px;
  opacity: 0.9; }
