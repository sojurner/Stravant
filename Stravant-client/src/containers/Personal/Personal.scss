.personal-stats {
  display: flex;
  justify-content: center;
}

path {
  cursor: pointer;
}

.personal-stats {
  position: absolute;
}

.legend {
  background-color: #f9f9f9;
  border: 1px solid #e5e5e5;
  border-radius: 12px;
  font-size: 0.8em;
  padding-top: 2rem;
}

.running-avg {
  text-align: center;
  font-family: daybold;
  font-size: 2.5rem;
  padding: auto;
  color: #a49188;
  margin: 0;
}

.pie-chart-wrap {
  display: flex;
  justify-content: center;
}

li {
  display: flex;
  left: 0.6rem;
}

.data-details {
  position: fixed;
  font-family: daybold;
  font-size: 1.5rem;
  color: #ffc1b8;
  letter-spacing: 2px;
  background: #c5696a;
  border-radius: 0.4em;
  margin: 26rem 0;
  padding: 1rem;
  opacity: 0.9;
}

.legend-container-fc25d28b8dda2e2172d549eafe0de054a26a4837 {
  width: 11rem;
  margin: -3rem 0 0 -23rem;
  position: fixed;
  top: 37%;
  left: 50%;
  padding: 0.5rem 0 0 0;
  font-family: daybold;
  font-size: 1.5rem;
}

.legend {
  padding-top: 0.7rem;
  background-color: #b3aaab;
}

.li {
  padding-top: 0.4rem;
}

.spinning-globe {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  margin: 3rem auto;
  background: center/cover url(../../images/earth.png);
  box-shadow: inset 0 0 25px #b07979;
  animation: world 35s linear infinite;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #a49188;
}

.spinning-globe:hover {
  box-shadow: inset 0 0 25px #8b4242;
}

@keyframes world {
  0% {
    background-position: 0 0;
  }
  100% {
    background-position: 719px 0;
  }
}

@media screen and (max-width: 800px) {
  .legend-container-fc25d28b8dda2e2172d549eafe0de054a26a4837 {
    width: 8.3rem;
    margin: -3rem 0 0 5rem;
    top: 85%;
    left: 0%;
    padding: 0.5rem 0 0 0;
    font-family: daybold;
    font-size: 1rem;
    z-index: 1;
  }
}

@media screen and (max-width: 510px) {
  .pie-chart-wrap {
    margin-top: 1.4rem;
  }

  .running-avg {
    margin: 0.7rem;
  }

  .data-details {
    margin: 27.5rem 0 0 0;
  }

  .pie-chart-7450ce7f65e862c33c80040ea675f203302df7d5 {
    position: absolute;
    left: 2rem;
  }
}
