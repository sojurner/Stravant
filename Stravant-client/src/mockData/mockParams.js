export const tokenParams = [
  'https://www.strava.com/oauth/token',
  {
    body:
      '{"client_id":28236,"client_secret":"829f12d9bdb8ce424edf87deff4551c117d3ed36","code":"r5423rffda4c234ctwtwert2ct"}',
    headers: { 'Content-Type': 'application/json' },
    method: 'POST'
  }
];

export const aggregateParams = [
  'https://www.strava.com/api/v3/athletes/28236/stats',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const weeklyParams = [
  'https://www.strava.com/api/v3/athlete/activities?before=1536123319&after=1536036919',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];
