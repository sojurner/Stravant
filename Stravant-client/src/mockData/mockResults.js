export const aggregateResult = { Biked: 0, Ran: 6.472964574269732, Swam: 0 };

export const tokenResult = {
  access_token: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  athlete: {
    badge_type_id: 0,
    city: null,
    country: null,
    created_at: '2018-09-04T22:53:48Z',
    email: 'pykim0591@gmail.com',
    firstname: 'Paul',
    follower: null,
    friend: null,
    id: 34515396,
    lastname: 'K',
    premium: false,
    profile:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    profile_medium:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    resource_state: 2,
    sex: 'M',
    state: null,
    summit: false,
    updated_at: '2018-09-07T12:20:05Z',
    username: null
  },
  token_type: 'Bearer'
};

export const weeklyResult = {
  Friday: 3784.3,
  Monday: 0,
  Saturday: 0,
  Sunday: 0,
  Thursday: 0,
  Tuesday: 0,
  Wednesday: 0
};
