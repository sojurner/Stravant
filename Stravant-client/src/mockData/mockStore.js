export const currentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    userId: 34515396,
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
  },
  totalStats: {},
  weeklyStats: {}
};

export const pomStatus = {
  pomStatus: true,
  lastPom: '',
  pomHistory: {
    'Tue, Sep 11, 2018 8:39 PM': { second: 19, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 8:40 PM': { second: 3, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 10:22 PM': { second: 3, minute: 0, hour: 0 }
  }
};

export const clubInfo = {
  clubInfo: { clubName: 'Stravant Club', clubId: 473825 },
  clubActivity: {
    mostRecent: {
      name: 'Pamela',
      distance: 4.270478558110628,
      activity: 'Morning Run'
    },
    Pamela: {
      totalDistance: 62.88999999999999,
      totalTime: 635,
      membersRecent: 'Afternoon Run'
    },
    Paul: { totalDistance: 6.46, totalTime: 78, membersRecent: 'Morning Run' },
    Jesse: {
      totalDistance: 277.78000000000003,
      totalTime: 5292,
      membersRecent: 'Morning Run'
    }
  }
};

export const mockStore = {
  ...currentUser,
  ...pomStatus,
  ...clubInfo
};
