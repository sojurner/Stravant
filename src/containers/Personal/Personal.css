.personal-stats {
  display: flex;
  justify-content: center; }

path {
  cursor: pointer; }

.personal-stats {
  position: absolute; }

.legend {
  background-color: #f9f9f9;
  border: 1px solid #e5e5e5;
  border-radius: 12px;
  font-size: 0.8em;
  padding: 2rem; }

.pie-chart-wrap {
  display: flex;
  justify-content: center; }

li {
  display: flex;
  left: 0.6rem; }
