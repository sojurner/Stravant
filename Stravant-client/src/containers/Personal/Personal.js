import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PieChart, Legend, ToolTip } from 'react-easy-chart';
import * as apiFetch from '../../helpers/apiCalls/apiCalls';
import * as userActions from '../../actions/userAction';
import './Personal.css';
import { type } from 'os';

export class Personal extends Component {
  constructor() {
    super();
    this.state = {
      showToolTip: false,
      top: '',
      left: '',
      value: '',
      key: '',
      goal: 0
    };
  }

  componentDidMount() {
    this.getUserStats();
  }

  getUserStats = async () => {
    const { setTotalStats } = this.props;
    const { accessToken, userId } = this.props.currentUser.info;
    const userStats = await apiFetch.getAggregateStats(accessToken, userId);
    setTotalStats(userStats);
  };

  mouseOverHandler = data => {
    console.log(data);
    this.setState({
      key: data.data.key,
      value: data.data.value,
      showToolTip: true
    });
  };

  mouseOutHandler = () => {
    this.setState({ showToolTip: false });
  };

  render() {
    const { totalStats } = this.props.currentUser;
    const filteredActivity = Object.keys(totalStats).map(type => {
      const destructured = totalStats[type];
      let dataArr = [];
      let configArr = [];
      Object.keys(destructured).forEach((stat, index) => {
        const filtered = {
          key: stat,
          value: Math.round(destructured[stat] * 100) / 100,
          color: `rgb(${index}04, 1${index}9, 2${index}4)`
        };

        const config = { color: `rgb(${index}04, 1${index}9, 2${index}4)` };
        dataArr.push(filtered);
        configArr.push(config);
      });
      return (
        <div>
          <PieChart
            className="personal-piechart"
            data={dataArr}
            innerHoleSize={150}
            mouseOverHandler={this.mouseOverHandler}
            mouseOutHandler={event => this.mouseOutHandler(event)}
            padding={10}
          />
          <Legend data={dataArr} config={configArr} dataId={'key'} />
        </div>
      );
    });

    return (
      <div className="pie-chart-wrap">
        {filteredActivity}
        {this.state.value &&
          this.state.showToolTip && (
            <p className="data-details">
              {`${this.state.key}: ${this.state.value}`}
            </p>
          )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentUser: state.currentUser
});

const mapDispatchToProps = dispatch => ({
  setTotalStats: stats => dispatch(userActions.setTotalStats(stats))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Personal);
