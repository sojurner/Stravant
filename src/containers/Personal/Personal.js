import React, { Component } from 'react';
import { PieChart } from 'react-easy-chart';
import './Personal.css';

class Personal extends Component {
  constructor() {
    super();
    this.state = {
      showToolTip: false,
      top: '',
      left: '',
      value: '',
      key: ''
    };
  }

  mouseOverHandler = (data, event) => {
    this.setState({
      showToolTip: true,
      top: event.y,
      left: event.x,
      value: data.value,
      key: data.data.key
    });
  };

  mouseMoveHandler = e => {
    if (this.state.showToolTip) {
      this.setState({ top: e.y, left: e.x });
    }
  };

  mouseOutHandler = () => {
    this.setState({ showToolTip: false });
  };

  render() {
    return (
      <div>
        <PieChart
          labels
          className="personal-piechart"
          data={[
            { key: 'Goal', value: 100, color: '#DC143C' },
            { key: 'Current', value: 300, color: '#aaac84' }
          ]}
          innerHoleSize={200}
          mouseOverHandler={this.mouseOverHandler}
          mouseOutHandler={event => this.mouseOutHandler(event)}
          mouseMoveHandler={event => this.mouseMoveHandler(event)}
          padding={10}
        />
        {this.state.showToolTip &&
          this.state.key === 'Current' && (
            <p
              className="personal-stats"
              top={this.state.top}
              left={this.state.left}
            >
              Your Pom-meter is {this.state.value} Miles!
            </p>
          )}
        {this.state.showToolTip &&
          this.state.key === 'Goal' && (
            <p
              className="personal-stats"
              top={this.state.top}
              left={this.state.left}
            >
              C'mon {this.state.value} miles left!!
            </p>
          )}
      </div>
    );
  }
}

export default Personal;
