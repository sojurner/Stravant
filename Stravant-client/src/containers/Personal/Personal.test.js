import React from 'react';
import { Personal, mapStateToProps, mapDispatchToProps } from './Personal';
import * as store from '../../mockData/mockStore';

jest.mock('../../helpers/apiCalls/apiCalls');

describe('Personal', () => {
  let wrapper;
  let mockCurrentUser;
  let mockSetTotal;

  beforeEach(() => {
    mockSetTotal = jest.fn();
    mockCurrentUser = store.currentUser;
    wrapper = shallow(
      <Personal currentUser={mockCurrentUser} setTotalStats={mockSetTotal} />
    );
  });

  it('should matchSnapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should matchSnapShot when there is totalStats', () => {
    mockCurrentUser = store.mockTotalStats;
    wrapper = shallow(
      <Personal currentUser={mockCurrentUser} setTotalStats={mockSetTotal} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match Mounted Snapshot', () => {
    wrapper = mount(
      <Personal currentUser={mockCurrentUser} setTotalStats={mockSetTotal} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should setState on mouseOverHandler', () => {
    wrapper.instance().mouseOverHandler({ data: { key: 1, value: 'paul' } });

    expect(wrapper.state('key')).toEqual(1);
    expect(wrapper.state('value')).toEqual('paul');
    expect(wrapper.state('showToolTip')).toEqual(true);
  });

  it('should setState on mouseOutHandler', () => {
    wrapper.instance().mouseOutHandler();

    expect(wrapper.state('showToolTip')).toEqual(false);
  });

  it('should call mouse overHandler on mouse over', () => {
    wrapper = mount(
      <Personal currentUser={mockCurrentUser} setTotalStats={mockSetTotal} />
    );
    const foundDiv = wrapper.find('.running-avg');

    expect(mouseOverHandler).toHaveBeenCalled();
  });

  it('should map to the store properly', () => {
    const mockStore = store.currentUser;
    const mapped = mapStateToProps(mockStore);

    expect(mapped.currentUser).toEqual(mockStore.currentUser);
  });

  it('should call dispatch function when when using a function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.setTotalStats();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
