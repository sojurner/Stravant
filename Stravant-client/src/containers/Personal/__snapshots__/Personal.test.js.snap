// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal should match Mounted Snapshot 1`] = `
<div
  className="pie-chart-wrap"
>
  <div
    className="spinning-globe"
  />
</div>
`;

exports[`Personal should matchSnapshot 1`] = `
<div
  className="pie-chart-wrap"
>
  <div
    className="spinning-globe"
  />
</div>
`;
